windshaft/driver.js

```diff
@@ -86,18 +86,20 @@ function getRequestParameters(request) {
         params.interactivity = 'uuid,occurred_from,data';
         params.style = eventsStyle;
 
-        // build query for record points
-        params.sql = baseRecordQuery;
-        if (params.id !== 'ALL') {
-            params.sql += filterRecordQuery + params.id + "'";
+        // build query for record points if do not already have a query
+        if (request.query.sql) {
+            params.sql = request.query.sql;
```

Comment:
Windshaft wraps all queries in a `SELECT`, which presumably would prevent actions other than reading. See #144, which should also add some controls via database roles. Windshaft is designed to allow arbitrary SQL queries.